libs/blocks/table/table.js

```diff
@@ -0,0 +1,462 @@
+        iconTag.tabIndex = 0;
+        sectionHeadTitle.appendChild(iconTag);
+
+        if (defaultExpandRow) {
+          iconTag.setAttribute('aria-expanded', 'true');
+          defaultExpandRow = false;
+        } else {
+          iconTag.setAttribute('aria-expanded', 'false');
+          let nextElement = row.nextElementSibling;
+          while (nextElement && !nextElement.classList.contains('divider')) {
+            nextElement.classList.add('hidden');
+            nextElement = nextElement.nextElementSibling;
+          }
+        }
+      }
+    } else if (previousRow && previousRow.querySelector('hr') && nextRow) {
+      nextRow.classList.add('section-row');
+      if (!isMerch) {
+        const sectionRowTitle = nextRow.querySelector('.col-1');
+        sectionRowTitle.classList.add('section-row-title');
+      }
+    } else if (!row.classList.contains('row-1') && (!isHighlightTable || !row.classList.contains('row-2'))) {
+      row.classList.add('section-row');
+      if (isMerch && !row.classList.contains('divider')) {
+        rowCols.forEach((merchCol) => {
+          merchCol.classList.add('col-merch');
+          const children = Array.from(merchCol.children);
+          const merchContent = createTag('div', { class: 'col-merch-content' });
+          if (children.length) {
+            children.forEach((child) => {
+              if (!child.querySelector('.icon')) {
+                merchContent.append(child);
+              }
+            });
+            merchCol.insertBefore(merchContent, merchCol.firstChild);
+          } else if (merchCol.innerText) {
+            const pTag = createTag('p', { class: 'merch-col-text' });
+            pTag.append(merchCol.innerText);
+            merchCol.innerText = '';
+            merchContent.append(pTag);
+            merchCol.append(merchContent);
+          }
+        });
+      } else {
+        const sectionRowTitle = row.querySelector('.col-1');
+        sectionRowTitle.classList.add('section-row-title');
+      }
+    }
+  });
+}
+
+function formatMerchTable(table) {
+  const rows = table.querySelectorAll('.row');
+  const rowsNum = rows.length;
+
+  const firstRow = rows[0];
+  const colsInRow = firstRow.querySelectorAll('.col');
+  const colsInRowNum = colsInRow.length;
+
+  for (let i = colsInRowNum; i > 0; i--) {
+    const cols = table.querySelectorAll(`.col-${i}`);
+    for (let j = rowsNum - 1; j >= 0; j--) {
+      const currentCol = cols[j];
+      if (!currentCol?.innerText && currentCol?.children.length === 0) {
+        currentCol.classList.add('no-borders');
+      } else {
+        currentCol?.classList.add('border-bottom');
+        break;
+      }
+    }
+  }
+}
+
+function handleMouseOut(cols) {
+  cols.forEach((e) => {
+    e.classList.remove('hover');
+    e.classList.remove('no-top-border');
+    e.classList.remove('hover-border-bottom');
+  });
+}
+
+function handleMouseOver(cols, table, colNum, isCollapseTable, lastSectionHead, lastExpandIcon) {
```

Comment:
This seems like a lot of params. Could it be paired down or would a destructured object be helpful?